rapids-cmake/cpm/nvcomp.cmake

```diff
@@ -0,0 +1,141 @@
+
+.. note::
+  Installation of nvcomp will occur if an INSTALL_EXPORT_SET is provided, and nvcomp
+  is added to the project via :cmake:command:`add_subdirectory <cmake:command:add_subdirectory>` by CPM.
+
+``USE_PROPRIETARY_BINARY``
+  By enabling this flag you and using the software, you agree to fully comply with the terms and conditions of
+  nvcomp's NVIDIA Software License Agreement. Found at https://developer.download.nvidia.com/compute/nvcomp/2.3/LICENSE.txt
+
+Result Targets
+^^^^^^^^^^^^^^
+  nvcomp::nvcomp target will be created
+
+Result Variables
+^^^^^^^^^^^^^^^^
+  :cmake:variable:`nvcomp_SOURCE_DIR` is set to the path to the source directory of nvcomp.
+  :cmake:variable:`nvcomp_BINARY_DIR` is set to the path to the build directory of nvcomp.
+  :cmake:variable:`nvcomp_ADDED`      is set to a true value if nvcomp has not been added before.
+  :cmake:variable:`nvcomp_VERSION`    is set to the version of nvcomp specified by the versions.json.
+  :cmake:variable:`nvcomp_proprietary_binary` is set to ON if the proprietary binary is being used
+
+#]=======================================================================]
+function(rapids_cpm_nvcomp)
+  list(APPEND CMAKE_MESSAGE_CONTEXT "rapids.cpm.nvcomp")
+
+  set(options)
+  set(one_value USE_PROPRIETARY_BINARY BUILD_EXPORT_SET INSTALL_EXPORT_SET)
+  set(multi_value)
+  cmake_parse_arguments(RAPIDS "${options}" "${one_value}" "${multi_value}" ${ARGN})
+
+  # Fix up RAPIDS_UNPARSED_ARGUMENTS to have EXPORT_SETS As this is need for rapids_cpm_find
+  if(RAPIDS_INSTALL_EXPORT_SET)
+    list(APPEND RAPIDS_UNPARSED_ARGUMENTS INSTALL_EXPORT_SET ${RAPIDS_INSTALL_EXPORT_SET})
+  endif()
+  if(RAPIDS_BUILD_EXPORT_SET)
+    list(APPEND RAPIDS_UNPARSED_ARGUMENTS BUILD_EXPORT_SET ${RAPIDS_BUILD_EXPORT_SET})
+  endif()
+
+  set(to_exclude OFF)
+  if(NOT RAPIDS_INSTALL_EXPORT_SET OR exclude)
+    set(to_exclude ON)
+  endif()
+
+  include("${rapids-cmake-dir}/cpm/detail/package_details.cmake")
+  rapids_cpm_package_details(nvcomp version repository tag shallow exclude)
+
+  # first see if we have a proprietary pre-built binary listed in versions.json and it if requested.
+  #
+  set(nvcomp_proprietary_binary OFF) # will be set to true by rapids_cpm_get_proprietary_binary
+  if(RAPIDS_USE_PROPRIETARY_BINARY)
+    include("${rapids-cmake-dir}/cpm/detail/get_proprietary_binary.cmake")
+    rapids_cpm_get_proprietary_binary(nvcomp ${version})
+  endif()
+
+  include("${rapids-cmake-dir}/cpm/find.cmake")
+  rapids_cpm_find(nvcomp ${version} ${RAPIDS_UNPARSED_ARGUMENTS}
+                  GLOBAL_TARGETS nvcomp::nvcomp
+                  CPM_ARGS
+                  GIT_REPOSITORY ${repository}
+                  GIT_TAG ${tag}
+                  GIT_SHALLOW ${shallow}
+                  EXCLUDE_FROM_ALL ${to_exclude}
+                  OPTIONS "BUILD_STATIC ON" "BUILD_TESTS OFF" "BUILD_BENCHMARKS OFF"
+                          "BUILD_EXAMPLES OFF")
+
+  # nvcomp creates the correct namespace aliases
```

Comment:
Then why do we need this block? Wouldn't needing this block indicate an improperly configured nvcomp?

Reply:
I'm still a bit confused by this one.

Reply:
When nvcomp is built from source it doesn't create a nvcomp::nvcomp alias just an add_library `nvcomp` target. rapids-cmake adds the alias so that all consumers of nvcomp just need a singular code path where they reference `nvcomp::nvcomp`